test/libsolidity/semanticTests/abiDecodeV1/dynamic_memory_copy.sol

```diff
@@ -0,0 +1,15 @@
+contract C {
+    function test(bytes memory buf) public view returns (bool same, bool inplaceDecoded) {
+        (uint256[] memory arr1, uint256[] memory arr2) = abi.decode(buf, (uint256[],uint256[]));
+        assembly {
+            same := eq(arr1, arr2)
+            inplaceDecoded := eq(arr1, add(buf, 0x60))
```

Comment:
Please add comments that explain what these flags are and what we should expect.

Reply:
Added extensive comments.